client/modules/datafiles/src/AddFileFolder/AddFileFolder.tsx

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import { Button } from 'antd';
+import DatafilesModal from '../DatafilesModal/DatafilesModal';
+import { useAuthenticatedUser, useFileListingRouteParams } from '@client/hooks';
+import styles from './AddFileFolder.module.css';
+
+export const AddFileFolder: React.FC = () => {
+  const { api, system, path } = useFileListingRouteParams();
+  const { user } = useAuthenticatedUser();
+  return (
+    <ul
+      style={{
+        border: '1px solid #e3e3e3',
+        listStyleType: 'none',
+        paddingLeft: '0px',
+      }}
+    >
+      {user && (
+        <>
+          <div style={{ margin: '0 10px 20px' }}>
+            <div className="btn-group btn-block">
+              <Button
+                className="btn btn-block btn-primary dropdown-toggle"
+                data-toggle="dropdown"
+              >
+                <i className="fa fa-plus-circle" role="none"></i> Add
+              </Button>
+              <ul className="dropdown-menu">
+                <li>
+                  <DatafilesModal.NewFolder
+                    api={api}
+                    system={system}
+                    path={path}
+                  >
+                    {({ onClick }) => (
+                      <a
+                        href="#"
+                        className={`${styles.navLink} ${styles.active}`}
+                        onClick={(e) => e.preventDefault()}
+                      >
+                        <span className="fa-stack fa-lg" onClick={onClick}>
+                          <i
+                            className="fa fa-folder fa-stack-2x"
+                            role="none"
+                          ></i>
+                          <i
+                            className="fa fa-plus fa-stack-1x fa-inverse"
+                            role="none"
+                          ></i>
+                        </span>
+                        <span onClick={onClick}>New Folder</span>
```

Comment:
antd has a `<Button type="link"/>` component that should be used when we need a "link" where we control the onClick behavior. This is an accessibility thing, since empty anchor tags and clickable spans are bad for keyboard/screenreader users.